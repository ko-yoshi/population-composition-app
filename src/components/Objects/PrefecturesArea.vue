<template>
  <fieldset class="prefectures">
    <legend>グラフに表示する都道府県を選択してください</legend>
    <template v-if="prefectures">
      <template v-for="i in prefectures.result" :key="i.prefCode">
        <PartsCheckboxBtn
          class="item"
          :label="i.prefName"
          :bg-color="getColorCode(i.prefCode)"
          @change="updatePrefCode(i.prefCode, i.prefName, $event)"
        />
      </template>
      <template v-if="prefectures.message">
        <p>データを取得できませんでした</p>
      </template>
    </template>
  </fieldset>
</template>
<script setup lang="ts">
  const { updatePrefCode, getColorCode } = usePopulationComposition()
  const { data: prefectures } = await useFetch('/api/prefectures')
</script>
<style lang="scss" scoped>
  .prefectures {
    @media screen and (max-width: 480px) {
      height: 200px;
      overflow-y: auto;
    }

    .item {
      margin: 0.2em;
    }
  }
</style>
