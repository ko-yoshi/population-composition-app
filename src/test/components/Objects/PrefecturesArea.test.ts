import { mount } from '@vue/test-utils'
import { expect, test, describe, vi } from 'vitest'
import PrefecturesArea from '~/components/Objects/PrefecturesArea.vue'

vi.mock('#app', () => ({}))
vi.mock('~/composition/usePopulationComposition', () => {
  return {
    __esModule: true,
    usePopulationComposition: vi.fn(() => {
      return {
        updatePrefCode: vi.fn(),
        getColorCode: vi.fn(),
      }
    }),
  }
})

vi.mock('~/api/prefectures', () => {
  return {
    __esModule: true,
    default: vi.fn(() => {
      return {
        data: {
          result: [
            { prefCode: '01', prefName: '北海道' },
            { prefCode: '02', prefName: '青森県' },
          ],
          message: '',
        },
      }
    }),
  }
})

describe('PrefecturesArea', () => {
  test('チェックボックスの確認', async () => {
    // const { data: prefectures } = await useFetch('/api/prefectures')の処理によりmountが失敗する
    // const wrapper = mount(PrefecturesArea)
    // await wrapper.vm.$nextTick()
    // const checkboxes = wrapper.findAll('.item')
    // expect(checkboxes.length).toBe(2)
    // expect(checkboxes[0].text()).toBe('北海道')
    // expect(checkboxes[1].text()).toBe('青森県')
  })

  test('snapshot', async () => {
    const wrapper = mount(PrefecturesArea)
    await wrapper.vm.$nextTick()
    expect(wrapper.element).toMatchSnapshot()
  })
})
