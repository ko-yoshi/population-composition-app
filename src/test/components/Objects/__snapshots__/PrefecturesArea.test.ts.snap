// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrefecturesArea > renders correctly 1`] = `null`;

exports[`PrefecturesArea > snapshot 1`] = `null`;
